src/python/dxpy/utils/executable_unbuilder.py

```diff
@@ -193,7 +226,30 @@ def untar_strip_leading_slash(tarfname, path):
 
                 untar_strip_leading_slash(fname, "resources")
                 os.unlink(fname)
-                deps_to_remove.append(dep)
+                # add dep name to deps_downloaded set
+                deps_downloaded.add(dep.get("name"))
+                
+            except DXError:
+                print("Download failed on dependent file with ID {} from region {}. Will try the next permitted region.\n".format(handler_id, region),
+                        file=sys.stderr)
+                # clean up deps already downloaded
+                deps_downloaded.clear()
+                created_resources_directory = False
+                shutil.rmtree("resources")
+                break
+            
+        # if all deps have been checked without an error, mark downloading as completed
+        else:
```

Comment:
There is a code block between `if` and `else` statements that is not indented, which is not very pythonic (not a good software development practice in general). What is the intended corresponding `if` to the `else` here? There is a code block between `if` and `else` statements that is not indented, which is not very pythonic (not a good software development practice in general).

Reply:
Oh it is part of the `for-else`. See Line 190.

Reply:
Oh, I see. This is an interesting construct. How about (generally) adding the `# no break` comment like in this explanation https://stackoverflow.com/a/23748240 to making it more clear what the `else` corresponds to?